src/libraries/System.Private.Xml/src/System/Xml/XmlCharType.cs

```diff
@@ -283,46 +228,4106 @@ private static bool InRange(int value, int start, int end)
             return (uint)(value - start) <= (uint)(end - start);
         }
 
-#if XMLCHARTYPE_GEN_RESOURCE
```

Comment:
can you leave the generating code in DEBUG and perhaps add a Debug.Assert comparing this array with what is generated? (just make sure it runs only once) In case we ever need to change this for some reason this won't be some magic array

Reply:
The generation code doesn't work. I started trying to use it but it's leftover from when the author of the current format was working with an older version of the data so it isn't useful. I ended up dumping the contents of the binary file rather than trying to work from that snippet.  If it had been useful I would have kept it but it isn't useful currently.

I asked in discord if people thought this was a good use for a source generator and the answer was that it's static data so it doesn't need generating each time. In either case static vs generated the time is going to be spent parsing it and that's unavoidable without going back to the resource load.